tests/swagger/test_swagger.py

```diff
@@ -831,6 +831,108 @@ def test_must_add_body_parameter_to_method_openapi_required_true(self):
+
+    def test_must_add_validator_parameters_to_method_with_validators_false(self):
+
+        self.editor.add_request_validator_to_method("/foo", "get", False, False)
+
+        expected = {"NO_VALIDATE": {"validateRequestBody": False, "validateRequestParameters": False}}
+
+        self.assertEqual(expected, self.editor.swagger["x-amazon-apigateway-request-validators"])
+        self.assertEqual(
+            "NO_VALIDATE", self.editor.swagger["paths"]["/foo"]["get"]["x-amazon-apigateway-request-validator"]
+        )
+
+    def test_must_add_validator_parameters_to_method_with_validators_mixing(self):
+
+        self.editor.add_request_validator_to_method("/foo", "get", True, False)
+
+        expected = {"BODY": {"validateRequestBody": True, "validateRequestParameters": False}}
+
+        self.assertEqual(expected, self.editor.swagger["x-amazon-apigateway-request-validators"])
+        self.assertEqual("BODY", self.editor.swagger["paths"]["/foo"]["get"]["x-amazon-apigateway-request-validator"])
+
+    def test_must_add_validator_parameters_to_method_and_not_duplicate(self):
+        self.original_swagger["paths"].update(
+            {
+                "/bar": {
+                    "get": {
+                        "x-amazon-apigateway-integration": {"test": "must have integration"},
+                        "parameters": [{"test": "existing parameter"}],
+                    }
+                },
+                "/foo-bar": {
+                    "get": {
+                        "x-amazon-apigateway-integration": {"test": "must have integration"},
+                        "parameters": [{"test": "existing parameter"}],
+                    }
+                },
+            }
+        )
+
+        editor = SwaggerEditor(self.original_swagger)
+
+        editor.add_request_validator_to_method("/foo", "get", True, True)
+        editor.add_request_validator_to_method("/bar", "get", True, True)
+        editor.add_request_validator_to_method("/foo-bar", "get", True, True)
+
+        expected = {"FULL": {"validateRequestBody": True, "validateRequestParameters": True}}
+
+        self.assertEqual(expected, editor.swagger["x-amazon-apigateway-request-validators"])
+        self.assertEqual("FULL", editor.swagger["paths"]["/foo"]["get"]["x-amazon-apigateway-request-validator"])
+        self.assertEqual("FULL", editor.swagger["paths"]["/bar"]["get"]["x-amazon-apigateway-request-validator"])
+        self.assertEqual("FULL", editor.swagger["paths"]["/foo-bar"]["get"]["x-amazon-apigateway-request-validator"])
+
+        self.assertEqual(1, len(editor.swagger["x-amazon-apigateway-request-validators"].keys()))
+
+    @parameterized.expand(
+        [
+            param(True, False, "BODY"),
+            param(True, True, "FULL"),
+            param(False, True, "REQUEST"),
+            param(False, False, "NO_VALIDATE"),
+        ]
+    )
+    def test_must_normalize_validator_names(self, validate_body, validate_request, normalized_name):
```

Comment:
You should maybe replace the `normalize` name with `return` here, no normalization is done.

Reply:
It seems it was left from the first review, nice catch 👍